### PR TITLE
Fix typo in transform.py from [w,h] to [h,w]

### LINKED ISSUE #193: [Question] Which is correct  [cx, cy, w, h, angle] or [cx, cy, h, w, angle] ?
In the implementation, it seems to have an object format such as [cx, cy, w, h, angle].
https://github.com/open-mmlab/mmrotate/blob/b78bab24dc35657fe4e349d8c033e15848101eef/mmrotate/core/bbox/transforms.py#L103.

But in the document [intro.md](https://github.com/open-mmlab/mmrotate/blob/main/docs/en/intro.md#what-is-rotated-box), it seems to explain as 
[x_center, y_center, height, width, theta].

Should the document be fixed?

### PR DESCRIPTION
**Pull request by warriors-30**
## Motivation
Motivated by an issue: [Question] Which is correct [cx, cy, w, h, angle] or [cx, cy, h, w, angle]?](https://github.com/open-mmlab/mmrotate/issues/193)
resolve #193 
There are typos in mmrotate/core/bbox/transform.py. 

## Modification
- switch the positions of all the w and h in pairs

## Checklist
1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmrotate/core/bbox/transforms.py [hunk 1]
```diff
@@ -246,7 +246,7 @@ def poly2obb_oc(polys):
         polys (torch.Tensor): [x0,y0,x1,y1,x2,y2,x3,y3]
 
     Returns:
-        obbs (torch.Tensor): [x_ctr,y_ctr,w,h,angle]
+        obbs (torch.Tensor): [x_ctr,y_ctr,h,w,angle]
```
Comment (yangxue0827) [inline mmrotate/core/bbox/transforms.py:249]: `_ws`, `_hs`, `ws`, and `hs` in this function should also be exchanged.

File: mmrotate/core/bbox/transforms.py [hunk 2]
```diff
@@ -272,7 +272,7 @@ def poly2obb_le135(polys):
...
```
Comment (yangxue0827) [inline mmrotate/core/bbox/transforms.py:275]: `width` and `height` in this function should also be exchanged.

File: mmrotate/core/bbox/transforms.py [hunk 3]
```diff
@@ -305,7 +305,7 @@ def poly2obb_le90(polys):
...
```
Comment (yangxue0827) [inline mmrotate/core/bbox/transforms.py:308]: `width` and `height` in this function should also be exchanged.

File: mmrotate/core/bbox/transforms.py [hunk 4]
```diff
@@ -364,7 +364,7 @@ def poly2obb_np_le135(poly):
         polys (ndarray): [x0,y0,x1,y1,x2,y2,x3,y3]
 
     Returns:
-        obbs (ndarray): [x_ctr,y_ctr,w,h,angle]
+        obbs (ndarray): [x_ctr,y_ctr,h,w,angle]
```
Comment (yangxue0827) [inline mmrotate/core/bbox/transforms.py:367]: `width` and `height` in this function should also be exchanged.

File: mmrotate/core/bbox/transforms.py [hunk 5]
```diff
@@ -397,57 +397,57 @@ def poly2obb_np_le90(poly):
...
+    p4x, p4y = x - hx + wx, y - hy + wy
     return torch.stack([p1x, p1y, p2x, p2y, p3x, p3y, p4x, p4y], dim=-1)
 
 
 def obb2poly_le135(rboxes):
     """Convert oriented bounding boxes to polygons.
 
     Args:
-        obbs (torch.Tensor): [x_ctr,y_ctr,w,h,angle]
+        obbs (torch.Tensor): [x_ctr,y_ctr,h,w,angle]
```
Comment (yangxue0827) [inline mmrotate/core/bbox/transforms.py:450]: `width` and `height` in this function should also be exchanged.

File: mmrotate/core/bbox/transforms.py [hunk 6]
```diff
@@ -734,7 +734,7 @@ def obb2poly_np_le135(rrects):
     """Convert oriented bounding boxes to polygons.
 
     Args:
-        obbs (ndarray): [x_ctr,y_ctr,w,h,angle,score]
+        obbs (ndarray): [x_ctr,y_ctr,h,w,angle,score]
```
Comment (yangxue0827) [inline mmrotate/core/bbox/transforms.py:737]: `width` and `height` in this function should also be exchanged.

File: mmrotate/core/bbox/transforms.py [hunk 7]
```diff
@@ -926,10 +926,10 @@ def gt2gaussian(target):
...
     edge_2 = target[:, 2, :] - target[:, 1, :]
-    w = (edge_1 * edge_1).sum(dim=-1, keepdim=True)
-    w_ = w.sqrt()
-    h = (edge_2 * edge_2).sum(dim=-1, keepdim=True)
-    diag = torch.cat([w, h], dim=-1).diag_embed() / (4 * L * L)
+    h = (edge_1 * edge_1).sum(dim=-1, keepdim=True)
+    h_ = h.sqrt()
+    w = (edge_2 * edge_2).sum(dim=-1, keepdim=True)
+    diag = torch.cat([h, w], dim=-1).diag_embed() / (4 * L * L)
     cos_sin = edge_1 / w_
```
Comment (yangxue0827) [inline mmrotate/core/bbox/transforms.py:933]: ` cos_sin = edge_1 / h_`

**Review by yangxue0827 — COMMENTED** [review]
Please confirm carefully and repeatedly, because once there is a missed or wrong modification, it will have very serious consequences for the repo, and it is difficult to locate the error.

**Comment by yangxue0827** [comment]
Please confirm carefully and repeatedly, because once there is a missed or wrong modification, it will have very serious consequences for the repo, and it is difficult to locate the error.

**Comment by yangxue0827** [comment]
Except for `transformer.py`, the docstrings of almost all related functions are annotated with `(x, y, w, h, theta)`. I would prefer to revise the documentation rather than the code.
@warriors-30

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @warriors-30 ！We are grateful for your efforts in helping improve mmrotate open-source project during your personal time. 
Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）
Thank you again for your contribution❤

**Pull request closed without merging**